iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/setup/ProvisioningCommon.java

```diff
@@ -305,6 +320,10 @@ public ProvisioningStatus registerDevice(ProvisioningDeviceClientTransportProtoc
                 assertFalse(deviceId.isEmpty());
                 assertFalse(provisionedHubUri.isEmpty());
 
+                if (jsonPayload != null || !jsonPayload.isEmpty())
+                {
+                    assertTrue(provisioningStatus.provisioningDeviceClientRegistrationInfoClient.getProvisioningPayload() == jsonPayload);
```

Comment:
>( [](start = 30, length = 1)

Add an argument to assertTrue to explain what it would mean if this failure happened assertTrue("Did not receive expected payload json", ..., ...)